### PR TITLE
pages/learn: Add fan profile and power mode information

### PR DESCRIPTION
**Pull request by acostach**
... for Jetson Orin devices running Jetpack 6

Change-type: patch

---

> *Please make sure to read the [CONTRIBUTING](https://github.com/balena-io/docs/blob/master/CONTRIBUTING.md) document before opening the PR for relevant information on contributing to the documentation. Thanks!*

### CODE REVIEW

File: pages/learn/develop/hardware/i2c-and-spi.md [hunk 1]
```diff
@@ -348,6 +350,16 @@ After the custom device tree has been validated, it can be included in newer bal
...
 
+### Configurable fan profiles
+
+Jetson Orin Devices running balenaOS revisions newer than <TODO> and supervisor versions greater than <TODO> offer support for configurable fan profiles. You can switch between the two options provided by Jetpack: "quiet" and "cool", by navigating to the "Device Configuration" page in the balena-cloud dashboard and selecting the desired option. The change is performed at runtime, and does not trigger a device reboot. Preloading the [fan profile configuration](/reference/OS/configuration/#fanprofile) before provisioning your device can also be achieved by editing [config.json](/reference/OS/configuration/#about-configjson)
```
[the remark this answers is not in the document]
Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md]: Thanks @cywang117 , I updated the section

File: pages/learn/develop/hardware/i2c-and-spi.md [hunk 2]
```diff
@@ -348,6 +350,16 @@ After the custom device tree has been validated, it can be included in newer bal
...
+
+### Configurable power modes
+
+The same balenOS versions for Jetson Orin which support configurable fan profiles also offer the possibility for selecting the desired power mode. You can choose between "low", "mid" and "high" or specify the power mode index directly by navigating to the "Device Configuration" page in the balena-cloud dashboard. The available power modes indexes for your device-type are visible in the host OS in `/etc/nvpmodel.conf`. Similar to the fan profile, preloading the desired power mode configuration can be achieved by editing [config.json](/reference/OS/configuration/#about-configjson) and specifying the desired [power mode](/reference/OS/configuration/#powermode) before provisioning the device.
```
[the remark this answers is not in the document]
Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md]: updated, thanks for pointing these out

File: pages/learn/develop/hardware/i2c-and-spi.md [hunk 3]
```diff
@@ -348,6 +350,16 @@ After the custom device tree has been validated, it can be included in newer bal
...
 
+### Configurable fan profiles
+
+Jetson Orin Devices running balenaOS revisions newer than <TODO> and supervisor versions greater than <TODO> offer support for configurable fan profiles. You can switch between the two options provided by Jetpack: "quiet" and "cool", by navigating to the "Device Configuration" page on the balenaCloud dashboard and selecting the desired option. The change is performed at runtime, and does not trigger a device reboot. Preloading the [fan profile configuration](/reference/OS/configuration/#fanprofile) before provisioning your device can also be achieved by editing [config.json](/reference/OS/configuration/#about-configjson)
```
Comment (vipulgupta2048) [inline pages/learn/develop/hardware/i2c-and-spi.md:355]: Selecting the desired option from how? Will it be a dropdown, will it be a checkbox?
What will the option be called on the Device Configuration page.

Comment (vipulgupta2048) [inline pages/learn/develop/hardware/i2c-and-spi.md:355]:
```suggestion
Jetson Orin Devices running balenaOS revisions newer than <TODO> and supervisor versions greater than <TODO> offer support for configurable fan profiles. You can switch between the options provided by Jetpack by navigating to the Device/Fleet Configuration tab on the sidebar in the balenaCloud dashboard and selecting the desired option. The change will be performed during runtime and doesn't trigger a device reboot. Preloading the [fan profile configuration](/reference/OS/configuration/#fanprofile) before provisioning your device can be achieved by editing the [config.json](/reference/OS/configuration/#about-configjson) file.
```

1. Let's not mention options mentioned in the docs on the account of Nvidia making a 3rd mode called "High power" and then that prompts us to correct our docs. Looking into the future this decreases our maintenance. 
2. Additionally, I am right to assume this is also a fleet configuration option?
3. Need to be specific at places.

File: pages/learn/develop/hardware/i2c-and-spi.md [hunk 4]
```diff
@@ -348,6 +350,16 @@ After the custom device tree has been validated, it can be included in newer bal
...
+
+### Configurable power modes
+
+The same balenaOS versions for Jetson Orin which support configurable fan profiles also offer the possibility for selecting the desired power mode. You can choose between "low", "mid" and "high" or specify the power mode index directly by navigating to the "Device Configuration" page on the balenaCloud dashboard. The available power modes indexes for your device type are visible in the host OS in `/etc/nvpmodel.conf`. Similar to the fan profile, preloading the desired power mode configuration can be achieved by editing [config.json](/reference/OS/configuration/#about-configjson) and specifying the desired [power mode](/reference/OS/configuration/#powermode) before provisioning the device.
```
Comment (vipulgupta2048) [inline pages/learn/develop/hardware/i2c-and-spi.md:359]:
```suggestion
The same balenaOS versions for Jetson Orin which support configurable fan profiles also offer the possibility for selecting the desired power mode. You can choose between "low", "mid" and "high" or specify the power mode index directly by navigating to the Device/Fleet Configuration tab on the sidebar in the balenaCloud dashboard. The available power modes indexes for your device type are visible in the host OS in `/etc/nvpmodel.conf`. Preloading the desired power mode configuration can be achieved by editing the [config.json](/reference/OS/configuration/#about-configjson) file and specifying the desired [power mode](/reference/OS/configuration/#powermode) before provisioning the device.
```

1. Being specific about which versions would support this feature in this section as well would be recommended. Sometimes we would only link this heading in support and the user won't be reading the heading above or understand what are configurable fan profiles. Specific is better.
2.

File: pages/learn/develop/hardware/i2c-and-spi.md [hunk 5]
```diff
@@ -348,6 +350,16 @@ After the custom device tree has been validated, it can be included in newer bal
...
+
+NOTE: In order for the selected configuration to be applied, the device will automatically reboot after you have chosen a different power mode.
```
Comment (vipulgupta2048) [inline pages/learn/develop/hardware/i2c-and-spi.md:361]: Better to add this to the section above only and not as a note.

**Review by vipulgupta2048 — CHANGES_REQUESTED** [review]

File: pages/learn/develop/hardware/i2c-and-spi.md (hunk 3 above)
Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md:355]: @cywang117 will this be a drop-down that also allows custom strings, or a string box?

Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md]: @vipulgupta2048  1) & 3) are addressed. @cywang117 can you please confirm if these configurations will be available for fleets as well?

File: pages/learn/develop/hardware/i2c-and-spi.md (hunk 4 above)
Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md]: 1) is addressed. I guess 2) was a typo or the comment was maybe not saved @vipulgupta2048 ?

File: pages/learn/develop/hardware/i2c-and-spi.md (hunk 5 above)
Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md]: done

File: pages/learn/develop/hardware/i2c-and-spi.md [hunk 6]
```diff
@@ -348,6 +350,14 @@ After the custom device tree has been validated, it can be included in newer bal
...
 
+### Configurable fan profiles
+
+Jetson Orin Devices running balenaOS revisions newer than <TODO> and supervisor versions greater than <TODO> offer support for configurable fan profiles. You can switch between the options provided by Jetpack by navigating to the Device/Fleet Configuration tab on the sidebar in the balenaCloud dashboard and selecting the desired option. The change will be performed during runtime and doesn't trigger a device reboot. Preloading the [fan profile configuration](/reference/OS/configuration/#fanprofile) before provisioning your device can be achieved by editing the [config.json](/reference/OS/configuration/#about-configjson) file.
```
Comment (cywang117) [inline pages/learn/develop/hardware/i2c-and-spi.md:355]:
```suggestion
Jetson Orin Devices running balenaOS revisions newer than v6.1.24 and supervisor versions greater than v16.10.0 offer support for configurable fan profiles. You can switch between the options provided by Jetpack by navigating to the Device/Fleet Configuration tab on the sidebar in the balenaCloud dashboard and selecting the desired option. The change will be performed during runtime and doesn't trigger a device reboot. Preloading the [fan profile configuration](/reference/OS/configuration/#fanprofile) before provisioning your device can be achieved by editing the [config.json](/reference/OS/configuration/#about-configjson) file.
```

File: pages/learn/develop/hardware/i2c-and-spi.md [hunk 7]
```diff
@@ -348,6 +350,14 @@ After the custom device tree has been validated, it can be included in newer bal
...
+
+### Configurable power modes
+
+Jetson Orin Devices running balenaOS revisions newer than <TODO> and supervisor versions greater than <TODO> also offer the possibility for selecting the desired power mode. You can choose between "low", "mid" and "high" or specify the power mode index directly by navigating to the Device/Fleet Configuration tab on the sidebar in the balenaCloud dashboard. Please note that your device(s) will automatically reboot to apply the new power mode. The available power modes indexes for your device type are visible in the host OS in `/etc/nvpmodel.conf`. Preloading the desired power mode configuration can be achieved by editing the [config.json](/reference/OS/configuration/#about-configjson) file and specifying the desired [power mode](/reference/OS/configuration/#powermode) before provisioning the device.
```
Comment (cywang117) [inline pages/learn/develop/hardware/i2c-and-spi.md:359]:
```suggestion
Jetson Orin Devices running balenaOS revisions newer than v6.1.24 and supervisor versions greater than v16.10.0 also offer the possibility for selecting the desired power mode. You can choose between "low", "mid" and "high" or specify the power mode index directly by navigating to the Device/Fleet Configuration tab on the sidebar in the balenaCloud dashboard. Please note that your device(s) will automatically reboot to apply the new power mode. The available power modes indexes for your device type are visible in the host OS in `/etc/nvpmodel.conf`. Preloading the desired power mode configuration can be achieved by editing the [config.json](/reference/OS/configuration/#about-configjson) file and specifying the desired [power mode](/reference/OS/configuration/#powermode) before provisioning the device.
```

File: pages/learn/develop/hardware/i2c-and-spi.md (hunk 3 above)
Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md]: @vipulgupta2048 it is a string box, I've extended the description in the last commit

Comment (acostach) [reply pages/learn/develop/hardware/i2c-and-spi.md]: These configuration options are available both at fleet and device level @vipulgupta2048

**Comment by acostach** [comment]
@vipulgupta2048 @cywang117 anything else to change on this PR, can we merge it?

**Review by cywang117 — APPROVED** [review]
I just rebased on top of master, LGTM!

**Review by vipulgupta2048 — DISMISSED** [review]

**Pull request merged**